### PR TITLE
Deleting a project, screen, etc. should delete all associated models

### PR DESCRIPTION
**Pull request by karthikv**
When a project is deleted, all of its associated screens should be deleted. When a screen is deleted, all of its associated components should be deleted, and so on and so forth.

### CODE REVIEW

**Comment by karthikv** [comment]
@seanmonstar Added commits.

**Comment by karthikv** [comment]
@tofumatt - @ednapiranha informed me that you might be available to review my code this week.  If you can, that'd be great! I think I've been putting too much on @seanmonstar thus far.

**Comment by tofumatt** [comment]
I can do it right now! (Ping me in #devecosystem on IRC next time you need an r?)

**Comment by seanmonstar** [comment]
Woops, sorry, I forgot about this.

File: lib/scaffold.js [hunk 1]
```diff
@@ -101,11 +124,74 @@ exports.generate = function(getKey, getDefault, allowedFields) {
...
     callback = callback || utils.noop;
+    id = String(id);
 
-    crud.remove(key, id, db, function(err, status) {
-      callback(err);
+    // set up the request to remove child scaffolds
+    var childReq = req;
+    var singularName = name.substring(0, name.length - 1);
+
+    if (!childReq.params)
```
Comment (tofumatt) [inline lib/scaffold.js]: I'd add some braces in here, even though it's single-line.

File: lib/scaffold.js [hunk 2]
```diff
@@ -18,9 +22,28 @@ var utils = require('./utils');
...
 
+  /* Get Hash Key
+   * Requires: web request
+   * Returns: database key for objects hash
+   */
+  function getHashKey(req) {
+    var key = req.session.email + ':';
+
+    var parentKeys = parents.map(function(parentName) {
+      var singularName = parentName.substring(0, parentName.length - 1);
```
Comment (tofumatt) [inline lib/scaffold.js:36]: I'm guessing this doesn't pluralize octopus, assuming I'm reading this code right.

Comment (karthikv) [inline lib/scaffold.js:36]: Well, `parents` is an array of plural scaffold names, so I assume you mean it doesn't singularize octopuses/octopi correctly. Regardless, I'm going to go ahead and assume that no scaffold will ever be called octopuses :P. Regardless, all that matters for a database key is understandability and consistency, which I believe are both maintained here.

Comment (tofumatt) [inline lib/scaffold.js:36]: Fair enough; sounds good.

**Pull request merged**